MapboxNavigation/NavigationMapView.swift

```diff
@@ -49,20 +38,60 @@ open class NavigationMapView: MGLMapView {
     
     public weak var navigationMapDelegate: NavigationMapViewDelegate?
     
-    override open func locationManager(_ manager: CLLocationManager!, didUpdateLocations locations: [CLLocation]!) {
-        guard let location = locations.first else { return }
-        
-        if let modifiedLocation = navigationMapDelegate?.navigationMapView?(self, shouldUpdateTo: location) {
-            super.locationManager(manager, didUpdateLocations: [modifiedLocation])
-        } else {
-            super.locationManager(manager, didUpdateLocations: locations)
+    public var userLocationForCourseTracking: CLLocation? {
+        didSet {
+            guard let location = userLocationForCourseTracking, CLLocationCoordinate2DIsValid(location.coordinate) else {
+                return
+            }
+            
+            // didUpdateLocationWithUserTrackingAnimated
+            // didUpdateLocationIncrementallyAnimated
+            // TODO: Call setVisibleCoordinateBounds instead to account for the location view’s bottom-weighted position on-screen.
+            let camera = MGLMapCamera(lookingAtCenter: location.coordinate, fromDistance: 300, pitch: 45, heading: location.course)
+            setCamera(camera, animated: true)
```

Comment:
For #370, we need to turn animation off, at least when the device is on battery power. If we leave animation on, we’d also need to enable a linear timing curve and stretch out the animation for one second to eliminate the jumping that currently occurs.